stdlib/ostruct.rb

```diff
@@ -60,10 +66,52 @@ def to_n
   def hash
     @table.hash
   end
+  
+  def table
+    @table
+  end
+  
+  def delete_field(name)
+    sym = name.to_sym
+    begin
+      singleton_class.__send__(:remove_method, sym, "#{sym}=")
+    rescue NameError
+    end
+    @table.delete sym
+  end
+  
+  def new_ostruct_member(name)
```

Comment:
Oh man, I just tested and the same happens in MRI, I thought it had some C-ext internal magic to avoid exploding when `new_ostruct_member` was redefined.

The `#new_ostruct_member` stuff is fine then, but `#table` doesn't seem to exist.

```ruby
>> x = OpenStruct.new
=> #<OpenStruct>
>> x.table
=> nil
>> x.y = 2
=> 2
>> x.table
=> nil
```